CHANGES.md

```diff
@@ -61,6 +62,7 @@
 - Fixes a bug under cygwin where dune-release was unable to find the commit hash corresponding to the release tag (#329, @gpetiot)
 - Fixes release names by explicitly setting it to match the released version (#338, @NathanReb)
 - Fix a bug that prevented release of a package whose version number contains invalid characters for a git branch. The git branch names are now sanitized. (#271, @gpetiot)
+- `publish`: Fix the process of inferring user name and repo from the dev repo uri (#318, @pitag-ha)
```

Comment:
Done. Since I need that commit for the `check` command to work, I've still left the commit in this PR for now and converted this PR to draft. I'll fix that and rebase once [the extracted PR](https://github.com/ocamllabs/dune-release/pull/348) is merged.